src/workerd/api/streams/standard.c++

```diff
@@ -1649,9 +1642,9 @@ struct ValueReadable final: private api::ValueQueue::ConsumerImpl::StateListener
     KJ_IF_SOME(s, state) {
       auto prp = js.newPromiseAndResolver<ReadResult>();
       s.consumer->read(js,
-          ValueQueue::ReadRequest{
+          kj::heap<ValueQueue::ReadRequest>({
```

Comment:
Why was kj::heap added here? Just making types more explicit?

Reply:
No, previously `ReadRequest` here was movable on it's own but the cleanup logic in the `~ReadRequest` was a bit problematic and was causing an UAF when moving to the new Queue/drain logic. Wrapping it in the kj::Own fixes that issue.

Reply:
I suppose an alternative is to implement `ReadRequest`'s move constructor/assignment operator correctly, e.g. invalidating its `byobReadRequest` when moved-from. Doing it the way you did it is probably more maintainable, though.